parcels/uxfieldset.py

```diff
@@ -0,0 +1,189 @@
+import cftime
+import numpy as np
+import uxarray as ux
+import cftime
+
+from parcels._compat import MPI
+from parcels._typing import GridIndexingType, InterpMethodOption, Mesh
+from parcels.field import DeferredArray, Field, NestedField, VectorField
+from parcels.grid import Grid
+from parcels.gridset import GridSet
+from parcels.particlefile import ParticleFile
+from parcels.tools._helpers import default_repr, fieldset_repr
+from parcels.tools.converters import TimeConverter, convert_xarray_time_units
+from parcels.tools.loggers import logger
+from parcels.tools.statuscodes import TimeExtrapolationError
+from parcels.tools.warnings import FieldSetWarning
+
+__all__ = ["UXFieldSet"]
+
+_inside_tol = 1e-6
+
+# class UXVectorField:
+#     def __init__(self, name: str, U: ux.UxDataArray, V: ux.UxDataArray, W: ux.UxDataArray | None = None):
+#         self.name = name
+#         self.U = U
+#         self.V = V
+#         self.W = W
+#         if self.W:
+#             self.vector_type = "3D"
+#         else:
+#             self.vector_type = "2D"
+
+#     def __repr__(self):
+#         return f"""<{type(self).__name__}>
+#     name: {self.name!r}
+#     U: {default_repr(self.U)}
+#     V: {default_repr(self.V)}
+#     W: {default_repr(self.W)}"""
+
+#     def eval(self, time, z, y, x, particle=None, applyConversion=True):
+        
+class UXFieldSet:
+    """A FieldSet class that holds hydrodynamic data needed to execute particles
+    in a UXArray.Dataset"""
+    # Change uxds to ds_list - which is a list of either uxDataset or xarray dataset
+    def __init__(self, uxds: ux.UxDataset, time_origin: float | np.datetime64 | np.timedelta64 | cftime.datetime = 0):
```

Comment:
the `uxds` argument will be changes to `ds` (standing for "DataSets"), which will be a `List(uxarray.UxDataset | xarray.Dataset)`